name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/SecuritiesChart.java

```diff
@@ -394,53 +409,206 @@ private void addSMAMarkerLines(int SMADays)
         lineSeriesSMA.setLineColor(
                         Display.getDefault().getSystemColor(SMADays == 200 ? SWT.COLOR_RED : SWT.COLOR_GREEN));
         lineSeriesSMA.setYAxisId(0);
+        lineSeriesSMA.setVisibleInLegend(true);
+
     }
 
     private void addInvestmentMarkerLines()
     {
+        int CounterBuy = 0;
```

Comment:
Hier machst Du Dir das Leben sehr schwer. Ich glaube es ist viel einfacher (und lesbarer) wenn Du *einmal* über die Buchungen iterierst. Da Du nicht weißt wie viele Buchungen es geben wird, sammelst Du die in ```List``` Objekten.

Und das Kriterium ist dann: wenn die List ```!txBuy.isEmpty()``` dann werden die Datenreihe dem Diagramm hinzugefügt.

Eine Liste kannst Du ganz einfach mit ```txBuy.toArray(new LocalDate[])``` in ein Array umwandeln und an ```TimelineChart#toJavaUtilDate``` weiterreichen.

Reply:
Wäre die andere Alternative nicht mittels ```HashMap``` nicht evtl. die bessere Alternative zu ```<list>```?
Dann könnte ich mittels ```hashMap.put("1", "foo")``` sowohl das Datum als auch Wert behalten und dann später gemäß deinem Hinweis mit ```.toArray``` wieder konvertieren.
``` hashMap.keySet().toArray(); // returns an array of keys```
``` hashMap.values().toArray(); // returns an array of values```